src/hooks/useSignMessage/useSignMessage.ts

```diff
@@ -271,11 +279,24 @@ const useSignMessage = ({
 
         await verifySignature(toSign, sig, requestedNetwork?.id)
 
-        resolve({ success: true, result: sig })
         addToast('Successfully signed!')
+
+        setSignedMessages( [...signedMessages, {
+            account: account.id,
+            networkId: requestedChainId,
+            date: new Date().getTime(),
+            typed: isTypedData,
+            signer: account.signer,
+            message: toSign.txn,
+            signature: sig,
+            dApp,
+          }]
+        )
+
         if (everythingToSign.length === 1) {
           !!onLastMessageSign && onLastMessageSign()
         }
+        resolve({ success: true, result: sig })
```

Comment:
You moved the `resolve()` part here, because on the web app it unmounts the web component and skips executing the logic afterwards, correct?

If so - please add a comment here. So that 1) we don't forget and 2) we give context to the other brave souls who will break or debug this one day.